src/gar.lua

```diff
@@ -75,26 +75,46 @@ local gar = {}
 --- @type Gateways
 GatewayRegistry = GatewayRegistry or {}
 
-GatewayRegistrySettings = GatewayRegistrySettings
-	or {
-		observers = {
-			maxPerEpoch = 50,
-			tenureWeightDays = 180,
-			tenureWeightPeriod = 180 * 24 * 60 * 60 * 1000, -- aproximately 180 days
-			maxTenureWeight = 4,
-		},
-		operators = {
-			minStake = 50000 * 1000000, -- 50,000 IO
-			withdrawLengthMs = 30 * 24 * 60 * 60 * 1000, -- 30 days to lower operator stake
-			leaveLengthMs = 90 * 24 * 60 * 60 * 1000, -- 90 days that balance will be vaulted
-			failedEpochCountMax = 30, -- number of epochs failed before marked as leaving
-			failedEpochSlashPercentage = 0.2, -- 20% of stake is returned to protocol balance
-		},
-		delegates = {
-			minStake = 500 * 1000000, -- 500 IO
-			withdrawLengthMs = 30 * 24 * 60 * 60 * 1000, -- 30 days
-		},
-	}
+--- @class ObserverSettings
+--- @field maxPerEpoch number
+--- @field tenureWeightDays number
+--- @field tenureWeightPeriod number
+--- @field maxTenureWeight number
+
+--- @class OperatorSettings
+--- @field minStake number
+--- @field withdrawLengthMs number
+--- @field leaveLengthMs number
+--- @field failedEpochCountMax number
+--- @field failedEpochSlashRate number
+
+--- @class DelegateSettings
+--- @field minStake number
+--- @field withdrawLengthMs number
+
+--- @class GatewayRegistrySettings
+--- @field observers ObserverSettings
+--- @field operators OperatorSettings
+--- @field delegates DelegateSettings
+GatewayRegistrySettings = {
+	observers = {
+		maxPerEpoch = 50,
+		tenureWeightDays = 180,
+		tenureWeightPeriod = 180 * 24 * 60 * 60 * 1000, -- approximately 180 days
+		maxTenureWeight = 4,
+	},
+	operators = {
+		minStake = 10000 * 1000000, -- 10,000 IO
```

Comment:
Nit: would feel a bit more comfortable to have 1000000 everywhere in the code set to a constant for mIOPerIO or something like that. It can be tricky to count the zeros visually beyond a small few.

Reply:
agree - i'll create.a ticket to standardrize all these